src/output/fmt.rs

```diff
@@ -79,15 +79,32 @@ impl Colorizer {
 
         let mut buffer = writer.buffer();
 
-        for piece in &self.pieces {
-            let mut color = ColorSpec::new();
-            color.set_fg(piece.1);
-            if piece.1 == Some(Color::Red) {
-                color.set_bold(true);
+        for (string, color) in &self.pieces {
+            if Colorizer::is_label(string) {
+                let mut label = string.clone();
+                label.pop(); // pops off the trailing colon ":".
+
+                let mut label_spec = ColorSpec::new();
+                label_spec.set_fg(*color);
+                label_spec.set_bold(true);
+                buffer.set_color(&label_spec)?;
+                buffer.write_all(label.as_bytes())?;
+
+                label_spec.set_fg(None); // the colon should be plane (with no color).
+                buffer.set_color(&label_spec)?;
+                buffer.write_all(":".as_bytes())?;
+            } else {
+                let mut color_spec = ColorSpec::new();
+                color_spec.set_fg(*color);
+
+                if *color == Some(Color::Red) {
+                    color_spec.set_bold(true);
+                }
+
+                buffer.set_color(&color_spec)?;
+                buffer.write_all(string.as_bytes())?;
```

Comment:
Well, this only works if we assume the text pieces doesn't have a trailing `:` I guess. What about bubbleing the logic up? Editing the `start_error() src/parse/errors.rs:400` to may also works.